### ISSUE TITLE
Not waiting for promises to be resolved in server-side tests

### ISSUE THREAD

**Issue by ratiofu** [practicalmeteor/mocha#44]
Server-side tests do not wait for a promise to be resolved. The following code reproduces the issue.

```
describe('some test suite', function() {
  it('fails on completion', function() {
    return new Promise((resolve, reject) => {
        reject(new Error('fails-just-fine'))
    })
  })
})
```

This will fail in the browser, like it should, but not on the server.

![screenshot of client and server test results](https://cdn.pbrd.co/images/1juWESsE.png)

I am running the tests with `meteor test --driver-package practicalmeteor:mocha --port 3100`

Specifically, it appears that in the [custom Runnable Setup](https://github.com/practicalmeteor/meteor-mocha/blob/meteor/meteor/src/setup/SetupRunnables.js#L113), the return value of the actual test method is entirely ignored:

```
func.call(self); // < returned promise is lost
done(); // < test immediately completes
```

My current work-around is to wire up the promise manually, but this is getting pretty verbose:

```
  it('fails on completion', function(done) {
    new Promise((resolve, reject) => {
        reject(new Error('fails-just-fine'))
    })
      .then(done)
      .catch(done)
  })
```

**Comment by mordka** [comment]
For a workaround you might want to catch in manually.

```
try {
  expect(1).to.equal(2);
} catch (err) {
  done(err);
}
done();
```

**Comment by ratiofu** [comment]
I don't have an issue with the exception not being caught per se. The problem is with returned promises not being handled properly. The same would be true if the test ran successfully: we wouldn't really know either way.

**Comment by rbabayoff** [comment]
@tjconsult did you check if mocha, independent of meteor, is supporting waiting for promises, if a promise is returned by a test case?

**Comment by ratiofu** [comment]
[It does.](https://mochajs.org/#working-with-promises)

**Comment by jsep** [comment]
@tjconsult I've just release a new release version of the package the solve this issue.

Try running: 
`meteor add practicalmeteor:mocha@=2.4.5-rc3.2`

Thanks :smile: !

**Comment by jsep** [comment]
This has been fix in 2.4.5_3 release

**Comment by ratiofu** [comment]
@jsep you rock. Thanks!
